applications/ParticleMechanicsApplication/custom_constitutive/flow_rules/mc_plastic_flow_rule.cpp

```diff
@@ -77,7 +77,7 @@ MCPlasticFlowRule::~MCPlasticFlowRule()
 {
 }
 
-void MCPlasticFlowRule::InitializeMaterial(YieldCriterionPointer pYieldCriterionPointer, HardeningLawPointer pHardeningPointer, const Properties& rProp)
+void MCPlasticFlowRule::InitializeMaterial(YieldCriterionPointer& pYieldCriterionPointer, HardeningLawPointer& pHardeningPointer, const Properties& rProp)
```

Comment:
Hello Pooyan, there is no special reason, I suppose I didn't want to make copies, but it is a shared pointer and I am not going to change what it is pointing inside the method. Therefore it is not necessary.